modules/instruments/php/module.class.inc

```diff
@@ -146,5 +146,37 @@ class Module extends \Module
         }
         return [];
     }
+
+    /**
+     * Return the data dictionary for all instruments installed on a LORIS
+     * instance.
+     *
+     * @param \LORIS\LorisInstance $loris The loris instance whose dictionary
+     *                                    should be retrieved
+     *
+     * @return \LORIS\Data\Dictionary\Category[]
+     */
+    public function getDataDictionary(\LORIS\LorisInstance $loris) : iterable
+    {
+        $DB = $loris->getDatabaseConnection();
+
+        $rows = $DB->pselectCol("SELECT Test_name FROM test_names", []);
```

Comment:
if the instrument uses `getFullName()` the test name would not be in the test_names table no ?

Reply:
nvm you would still need it for flag and battery and what not

Reply:
I think the test_names is the closest we have to an "instruments" table to list the installed instruments.. as you say, flag joins against it and I don't think we support anything showing up on the instrument_list page unless it's in the test_names table. (It's also how the instrument manager gets a list of instruments..)